src/app/view-pet/view-pet.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Location } from '@angular/common';
+
+import { Pet } from '../pet';
+import { PetService } from '../pet.service';
+
+@Component({
+  selector: 'app-view-pet',
+  templateUrl: './view-pet.component.html',
+  styleUrls: ['./view-pet.component.css']
+})
+export class ViewPetComponent implements OnInit {
+
+  private id: number;
+  pet!: Pet;
+  displayVetRecords: boolean=false;
+
+  constructor(private petService: PetService, private route: ActivatedRoute, private location: Location) {
+    this.id = Number(this.route.snapshot.paramMap.get('id'));
+   }
+
+  ngOnInit(): void {
+    this.getPet(this.id);
+  }
+
+  getPet(id: number): void {
+    this.petService.getPet(id).subscribe(pet => this.pet = pet);
```

Comment:
this is a subscription without an unsubscribe. I wanted to give an example of good separation of calls and a flow.

### Component Level
```typescript
delete(): Observable<void> {
    return this.usersService.deleteUser(this.selectedUserId)
	.pipe(
	    take(1),
	    tap(() => {
	        this.router.navigate(['/Users']);
	    })
        );
 }
```
Notice that the component level controls the lifecycle of the subscription by using take 1. This ensure the subscription is complete once it executes exactly one time.

### Service Level
This layer is sometimes optional depending on if you have business logic to perform. In my example I'll just do a pass through to an api service.

```typescript
public deleteUser(userId: number): Observable<void> {
    return this.api.delete(userId);
}
```
If you had business logic or mapping you wanted to do you would pipe it in here.

### API or Domain level service
```typescript
public delete(id: string): Observable<void> {
    return this.http.delete(`${BaseUrl}/user/${id}`, { headers: this.getHeaders() })
        .pipe(
            catchError(error => this.handleUnauthorizedRequest(error))
	);
}

Here, you do more like low level error handling or mapping but always returning that observable.
```

Reply:
Generally I think the only thing you're missing is the take 1 from the component but I wanted to show you the whole flow.